Jenkinsfile

```
@@ -45,36 +45,75 @@ node {
         }
 
-        stage('Download tools cert') {
-            // Download the certificate file and key file, used for signing the JFrog CLI binary.
-            sh """#!/bin/bash
-               builder/jfrog rt dl installation-files/certificates/jfrog/ --url https://entplus.jfrog.io/artifactory --flat --access-token=$DOWNLOAD_SIGNING_CERT_ACCESS_TOKEN
-                """
-
-            sh 'tar xvzf jfrogltd_signingcer_full.tar.gz'
-        }
-
         if ("$EXECUTION_MODE".toString().equals("Publish packages")) {
+            buildRpmAndDeb(version, architectures)
+
             stage('Npm Publish') {
-                print "publishing npm package"
                 publishNpmPackage(jfrogCliRepoDir)
             }
 
             stage('Build and Publish Docker Image') {
                 buildPublishDockerImage(version, jfrogCliRepoDir)
             }
         } else if ("$EXECUTION_MODE".toString().equals("Build CLI")) {
+            downloadToolsCert()
             print "Uploading version $version to Bintray"
             uploadCli(architectures)
         }
     }
 }
 
+def downloadToolsCert() {
+    stage('Download tools cert') {
+        // Download the certificate file and key file, used for signing the JFrog CLI binary.
+        sh """#!/bin/bash
+            builder/jfrog rt dl installation-files/certificates/jfrog/ --url https://entplus.jfrog.io/artifactory --flat --access-token=$DOWNLOAD_SIGNING_CERT_ACCESS_TOKEN
+            """
+
+        sh 'tar xvzf jfrogltd_signingcer_full.tar.gz'
+    }
+}
+
+def buildRpmAndDeb(version, architectures) {
+    boolean built = false
+    for (int i = 0; i < architectures.size(); i++) {
+        def currentBuild = architectures[i]
+        if (currentBuild.debianImage) {
+            stage("Build debian ${currentBuild.pkg}") {
+                build(currentBuild.goos, currentBuild.goarch, currentBuild.pkg, 'jfrog')
+                dir("$jfrogCliRepoDir") {
+                    sh "build/deb_rpm/build-scripts/pack.sh -b jfrog -v $version -f deb --deb-arch $currentBuild.debianArch --deb-build-image $currentBuild.debianImage"
+                    built = true
+                }
+            }
+        }
+        if (currentBuild.rpmImage) {
+            stage("Build rpm ${currentBuild.pkg}") {
+                build(currentBuild.goos, currentBuild.goarch, currentBuild.pkg, 'jfrog')
+                dir("$jfrogCliRepoDir") {
+                    sh "build/deb_rpm/build-scripts/pack.sh -b jfrog -v $version -f rpm --rpm-build-image $currentBuild.rpmImage"
+                    built = true
+                }
+            }
+        }
+    }
+
+    if (built) {
+        stage("Deploy deb and rpm") {
+            options = "--url https://releases.jfrog.io/artifactory --flat --access-token=$DEB_RPM_DEPLOY_ACCESS_TOKEN"
+            sh """#!/bin/bash
+                builder/jfrog rt u $jfrogCliRepoDir/build/deb_rpm/*.i386.deb jfrog-debs/pool/jfrog-cli/ --deb=xenial,bionic,eoan/contrib/i386 $options
```

Lets add latest Ubuntu as well
```suggestion
                builder/jfrog rt u $jfrogCliRepoDir/build/deb_rpm/*.i386.deb jfrog-debs/pool/jfrog-cli/ --deb=xenial,bionic,eoan,focal/contrib/i386 $options
```